Mathlib/Algebra/AffineMonoid/Basic.lean

```diff
@@ -0,0 +1,26 @@
+/-
+Copyright (c) 2025 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+module
+
+public import Mathlib.GroupTheory.Finiteness
+
+/-!
+# Affine monoids
+
+This file defines affine monoids as finitely generated cancellative torsion-free commutative
+monoids.
+-/
+
+public section
```

Comment:
When do you need `@[expose]` and when do you not?

Reply:
`@[expose]` only applies to `def`s. Whether you have it or not depends on if you think your `def` should be unfoldable (basically what was previously `irreducible_def`). Here (or I guess the next file along) I am very happy for the body `embedding` to be private, since the embedding is arbitrary.

Reply:
No, `abbrev`s always expose their body